Index: oscam-reader.c
===================================================================
--- oscam-reader.c	(revision 10102)
+++ oscam-reader.c	(working copy)
@@ -507,6 +507,13 @@
 
 void hexserial_to_newcamd(uchar *source, uchar *dest, uint16_t caid)
 {
+	if(caid == 0x5581 || caid == 0x4aee)    // Bulcrypt
+	{
+		dest[0] = 0x00;
+		dest[1] = 0x00;
+		memcpy(dest + 2, source, 4);
+		return;
+	}
 	caid = caid >> 8;
 	if(caid == 0x17 || caid == 0x06)    // Betacrypt or Irdeto
 	{
@@ -534,6 +541,13 @@
 
 void newcamd_to_hexserial(uchar *source, uchar *dest, uint16_t caid)
 {
+	if(caid == 0x5581 || caid == 0x4aee)    // Bulcrypt
+	{
+		memcpy(dest, source + 2, 4);
+		dest[4] = 0x00;
+		dest[5] = 0x00;
+		return;
+	}
 	caid = caid >> 8;
 	if(caid == 0x17 || caid == 0x06)    // Betacrypt or Irdeto
 	{
