--- module-camd35.c.orig        2014-12-26 21:27:49.548192114 +0200
+++ module-camd35.c     2014-12-26 21:29:23.800191708 +0200
@@ -331,9 +331,9 @@
        uint16_t au_caid = aureader->caid;

        // Bulcrypt has 2 caids and aureader->caid can't be used.
-       // Use 5581 for AU.
+       // Use ECM_REQUEST caid for AU.
        if(!au_caid && (er->caid == 0x5581 || er->caid == 0x4aee))
-               { au_caid = 0x5581; }
+               { au_caid = er->caid; }

        time(&now);
        if(!memcmp(cl->lastserial, aureader->hexserial, 8))
@@ -346,7 +346,7 @@
        {
                cl->disable_counter = 0;
                cs_log("%s emm-request sent (reader=%s, caid=%04X, auprovid=%06X)",
-                          username(cur_client()), aureader->label, aureader->caid,
+                          username(cur_client()), aureader->label, au_caid,
                           aureader->auprovid ? aureader->auprovid : b2i(4, aureader->prid[0]));
        }
        else if(cl->disable_counter > 2)
@@ -379,13 +379,6 @@
                mbuf[39] = 1;                           // no. caids
                mbuf[20] = au_caid >> 8;        // caid's (max 8)
                mbuf[21] = au_caid & 0xff;
-               if(au_caid == 0x5581)
-               {
-                       // Bulcrypt have two CAIDs, add the second one
-                       mbuf[39]++;
-                       mbuf[22] = 0x4aee >> 8;
-                       mbuf[23] = 0x4aee & 0xff;
-               }
                memcpy(mbuf + 40, aureader->hexserial, 6);  // serial now 6 bytes
                mbuf[47] = aureader->nprov;
                for(i = 0; i < aureader->nprov; i++)
--- module-cccam.c.orig 2014-12-27 02:49:06.214192264 +0200
+++ module-cccam.c      2014-12-27 02:36:11.545193090 +0200
@@ -1063,8 +1063,6 @@
                ofs = 2;
                break;
        }
-       if(caid == 0x5581 || caid == 0x4AEE)  //BULCRYPT:
-               { ofs = 0; }
        return ofs;
 }

@@ -1114,6 +1112,15 @@
        //
        //  //Place here your own adjustments!
        //}
+
+       if (caid == 0x5581 || caid == 0x4aee) { // Bulcrypt fix
+               out[4] = in[0];
+               out[5] = in[1];
+               out[6] = in[2];
+               out[7] = in[3];
+               return;
+       }
+
        hexserial_to_newcamd(in, tmp + 2, caid);
        UA_right(tmp, out, 8);
 }
@@ -1143,7 +1150,7 @@
                out[0] = in[4];
                out[1] = in[5];
                out[2] = in[6];
-               out[3] = in[7]; // hexbase
+               out[3] = in[7];
                return;
        }

--- oscam-reader.c.orig 2014-12-26 21:27:48.533192774 +0200
+++ oscam-reader.c      2014-12-26 21:29:23.801191819 +0200
@@ -541,6 +541,13 @@

 void newcamd_to_hexserial(uchar *source, uchar *dest, uint16_t caid)
 {
+       if(caid == 0x5581 || caid == 0x4aee)    // Bulcrypt
+       {
+               memcpy(dest, source + 2, 4);
+               dest[4] = 0x00;
+               dest[5] = 0x00;
+               return;
+       }
